### ISSUE TITLE
Random backend isn't random enough.

### ISSUE THREAD

**Issue by Tilps** [LeelaChessZero/lc0#67]
I think that for generating a large initial sets of positions each game should be played between random backends each of which have a different randomly selected seed to their hash.
(Can still have a default of 0 seed for producing predictable output.)

**Comment by mooskagh** [comment]
Random backend was not intended for generate bootstrap games, and didn't intend to have high quality random properties.
Instead it was written for performance benchmarking of MCTS (without GPU, mcts part is visible very well in profiler), and intended to be as fast as possible, while returning the same output for the same input.

If we really need proper random backend, we need to define what random actually means.

**Comment by Tilps** [comment]
On further consideration, its probably reasonable as is - so long as its used with standard training parameters (noise, temperature=1) - although it might be nice to have a mix of different seeds to provide a bit more variety, it probably doesn't need to be every game, or different opponents having different seeds.
(Adding a seed shouldn't affect the speed in any significant way I would hope.)

**Comment by Videodr0me** [comment]
I also noticed some funkyness in the randomness of games. I looked briefly at the generator functions and it seemed threadsafe but i did not dig deeper. Is the seed really the same for each simulated player in lc0? I just assumed it was generated by random out of some "physical" source (e.g. system clock) and is different for each "player" (in selfplay and/or training). If not we should really change that, as reusing the same random sequence cannot be good.
